cpp/src/plasma/lib/java/org_apache_arrow_plasma_PlasmaClientJNI.cc

```diff
@@ -0,0 +1,314 @@
+  plasma::ObjectID oid;
+  jbyteArray_to_object_id(env, object_id, &oid);
+
+  ARROW_CHECK_OK(client->Seal(oid));
+}
+
+JNIEXPORT void JNICALL Java_org_apache_arrow_plasma_PlasmaClientJNI_release(
+    JNIEnv* env, jclass cls, jlong conn, jbyteArray object_id) {
+  plasma::PlasmaClient* client = (plasma::PlasmaClient*)conn;
+  plasma::ObjectID oid;
+  jbyteArray_to_object_id(env, object_id, &oid);
+
+  ARROW_CHECK_OK(client->Release(oid));
+}
+
+JNIEXPORT jobjectArray JNICALL Java_org_apache_arrow_plasma_PlasmaClientJNI_get(
+    JNIEnv* env, jclass cls, jlong conn, jobjectArray object_ids, jint timeout_ms) {
+  plasma::PlasmaClient* client = (plasma::PlasmaClient*)conn;
+
+  jsize num_oids = env->GetArrayLength(object_ids);
+  std::vector<plasma::ObjectID> oids(num_oids);
+  std::vector<plasma::ObjectBuffer> obufs(num_oids);
+  for (int i = 0; i < num_oids; ++i) {
+    jbyteArray_to_object_id(env, (jbyteArray)env->GetObjectArrayElement(object_ids, i),
+                            &oids[i]);
+  }
+  // TODO: may be blocked. consider to add the thread support
+  ARROW_CHECK_OK(client->Get(oids.data(), num_oids, timeout_ms, obufs.data()));
+
+  jclass clsByteBuffer = env->FindClass("java/nio/ByteBuffer");
+  jclass clsByteBufferArray = env->FindClass("[Ljava/nio/ByteBuffer;");
+
+  jobjectArray ret = env->NewObjectArray(num_oids, clsByteBufferArray, NULL);
+  jobjectArray o = NULL;
+  jobject dataBuf, metadataBuf;
+  for (int i = 0; i < num_oids; ++i) {
+    o = env->NewObjectArray(2, clsByteBuffer, NULL);
+    if (obufs[i].data && obufs[i].data->size() != -1) {
+      dataBuf = env->NewDirectByteBuffer(const_cast<uint8_t*>(obufs[i].data->data()),
+                                         obufs[i].data->size());
+      if (obufs[i].metadata && obufs[i].metadata->size() > 0) {
+        metadataBuf = env->NewDirectByteBuffer(
+            const_cast<uint8_t*>(obufs[i].metadata->data()), obufs[i].metadata->size());
+      } else {
+        metadataBuf = NULL;
+      }
+    } else {
+      dataBuf = NULL;
+      metadataBuf = NULL;
+    }
+
+    env->SetObjectArrayElement(o, 0, dataBuf);
+    env->SetObjectArrayElement(o, 1, metadataBuf);
+    env->SetObjectArrayElement(ret, i, o);
+  }
+  return ret;
+}
+
+JNIEXPORT jboolean JNICALL Java_org_apache_arrow_plasma_PlasmaClientJNI_contains(
+    JNIEnv* env, jclass cls, jlong conn, jbyteArray object_id) {
+  plasma::PlasmaClient* client = (plasma::PlasmaClient*)conn;
+  plasma::ObjectID oid;
+  jbyteArray_to_object_id(env, object_id, &oid);
+
+  bool has_object;
+  ARROW_CHECK_OK(client->Contains(oid, &has_object));
+
+  if (has_object) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+JNIEXPORT void JNICALL Java_org_apache_arrow_plasma_PlasmaClientJNI_fetch(
+    JNIEnv* env, jclass cls, jlong conn, jobjectArray object_ids) {
+  plasma::PlasmaClient* client = (plasma::PlasmaClient*)conn;
+  jsize num_oids = env->GetArrayLength(object_ids);
+
+  /*
```

Comment:
remove unused code

If this is something that needs to be handled in the future, then add a comment with a TODO